crypto/evp/pmeth_lib.c

```diff
@@ -747,7 +747,7 @@ static int legacy_ctrl_to_param(EVP_PKEY_CTX *ctx, int keytype, int optype,
                   EVP_R_OPERATION_NOT_SUPPORTED_FOR_THIS_KEYTYPE);
         return -2;
     }
-    return 0;
+    return -2;
```

Comment:
hmm.. It looks like the only way this function runs is when new provider code is running..
if ((EVP_PKEY_CTX_IS_DERIVE_OP(ctx) && ctx->op.kex.exchprovctx != NULL)
            || (EVP_PKEY_CTX_IS_SIGNATURE_OP(ctx)
                && ctx->op.sig.sigprovctx != NULL)
            || (EVP_PKEY_CTX_IS_ASYM_CIPHER_OP(ctx)
                && ctx->op.ciph.ciphprovctx != NULL))

So does it need to be a legacy ctrl?
Or could be just use EVP_PKEY_CTX_set_params() instead?

Reply:
Yes, EVP_PKEY_CTX_set_params seems relevant here.

Reply:
Rolled back